app/controllers/concerns/metadata.rb

```diff
@@ -15,8 +15,10 @@ def set_headers
     def metadata(opts = {})
       opts[:total] ||= resolve_collection.count
       options = {}
-      options[:meta] = { total: opts[:total], search: params[:search],
-                         limit: pagination_limit, offset: pagination_offset }
+      options[:meta] = {
+        total: opts[:total], search: params[:search], tags: params[:tags],
```

Comment:
@akofink I'm not entirely sure about including this in the response and if yes, I don't know if we should use the parsed or the original format (my bet is on the parsed one)...

Reply:
It IS an array, but a but the tag itself is not just a simple string, but a hash with 3 elements (namespace, key, value) so the result right now is something like:
```
    "links": {
        "first": "/api/compliance/profiles?limit=10&offset=1&search=external%3Dfalse+and+canonical%3Dfalse",
        "last": "/api/compliance/profiles?limit=10&offset=2&search=external%3Dfalse+and+canonical%3Dfalse",
        "next": "/api/compliance/profiles?limit=10&offset=2&search=external%3Dfalse+and+canonical%3Dfalse"
    },
    "meta": {
        "limit": 10,
        "offset": 1,
        "tags": [
            {"namespace": "insights", "key": "environment", "value": "development"},
            {"namespace": "insights", "key": "environment", "value": "production"},
        ]
        "search": "external=false and canonical=false",
        "total": 15
    }
```

> I think it makes sense to include here, only for the systems endpoint, though

I think a simple `compact` can solve this problem and keep the complexity low (i.e. no tests for the hosts controller), we would show the tags only when they're passed...wdyt?

Reply:
I'm fine with the JSON object per tag. I'd include tags in the meta section even if no tags are specified, the same way we handle the other meta fields. That way, we communicate the default behavior (right now, default tags are just `[]`, but maybe in the future we will return hosts tagged with `compliance` or similar by default).

Reply:
Yeah, I've been trying point out that I would like to avoid a `tags if controller == 'Systems'` kind of condition in a global scope like the `Metadata` module :shrug: 

So do you have any preference on how to do this? I have too many ideas and I don't like any of them :sweat_smile:

Reply:
Nevermind, figured it out :yawning_face: mornings are hard :sweat_smile: